<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Content-Centric Networking in Java: org.ccnx.ccn.profiles.ccnd.CCNDaemonException Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.6.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="namespaces.html"><span>Packages</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><b>org</b>.<b>ccnx</b>.<b>ccn</b>.<b>profiles</b>.<b>ccnd</b>.<a class="el" href="classorg_1_1ccnx_1_1ccn_1_1profiles_1_1ccnd_1_1_c_c_n_daemon_exception.html">CCNDaemonException</a>
  </div>
</div>
<div class="contents">
<h1>org.ccnx.ccn.profiles.ccnd.CCNDaemonException Class Reference</h1><!-- doxytag: class="org::ccnx::ccn::profiles::ccnd::CCNDaemonException" --><!-- doxytag: inherits="org::ccnx::ccn::CCNException" -->
<p>A real AccessDeniedException doesn't appear in Java until 1.7.  
<a href="#_details">More...</a></p>

<p><a href="classorg_1_1ccnx_1_1ccn_1_1profiles_1_1ccnd_1_1_c_c_n_daemon_exception-members.html">List of all members.</a></p>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Package Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7c8eaa5c14d537a28c05d11648a0294c"></a><!-- doxytag: member="org::ccnx::ccn::profiles::ccnd::CCNDaemonException::CCNDaemonException" ref="a7c8eaa5c14d537a28c05d11648a0294c" args="(String message)" -->
&nbsp;</td><td class="memItemRight" valign="bottom"><b>CCNDaemonException</b> (String message)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a4989d456da1088d7f3103ce05e4625d6"></a><!-- doxytag: member="org::ccnx::ccn::profiles::ccnd::CCNDaemonException::CCNDaemonException" ref="a4989d456da1088d7f3103ce05e4625d6" args="(String message, Throwable cause)" -->
&nbsp;</td><td class="memItemRight" valign="bottom"><b>CCNDaemonException</b> (String message, Throwable cause)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="afe0fef0bd4c0ff92c7125f5d94aebd85"></a><!-- doxytag: member="org::ccnx::ccn::profiles::ccnd::CCNDaemonException::CCNDaemonException" ref="afe0fef0bd4c0ff92c7125f5d94aebd85" args="(Throwable cause)" -->
&nbsp;</td><td class="memItemRight" valign="bottom"><b>CCNDaemonException</b> (Throwable cause)</td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>A real AccessDeniedException doesn't appear in Java until 1.7. </p>
<p>Till then, make our own. Even then so far it doesn't take sub-exceptions as constructor arguments.</p>
<p>Similarly, IOException doesn't take a Throwable as a constructor argument till 1.6. Until we move to 1.6, fake one out. </p>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>src/org/ccnx/ccn/profiles/ccnd/CCNDaemonException.java</li>
</ul>
</div>
<hr size="1"/><address style="text-align: right;"><small>Generated on Thu Nov 4 13:34:29 2010 for Content-Centric Networking in Java by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.1 </small></address>
</body>
</html>
